pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -534,6 +553,55 @@ func (c *Cluster) DeployPrometheusRule(name, namespace string) error {
 	return nil
 }
 
+func (c *Cluster) templateToPrometheusRule(name, templateData string) (*monitoringv1.PrometheusRule, error) {
+	var rule monitoringv1.PrometheusRule
+	customPrometheusRule, err := c.computeCustomizedPrometheusAlerts()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to compute customized prometheus rule")
+	}
+	t, err := loadTemplate(name, templateData, customPrometheusRule)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load prometheus rule template")
+	}
+	err = yaml.Unmarshal(t, &rule)
+	if err != nil {
+		return nil, err
+	}
+	return &rule, nil
+}
+
+func loadTemplate(name, templateData string, p map[string]*cephv1.CephAlert) ([]byte, error) {
+	var writer bytes.Buffer
+	t := template.New(name)
+	t, err := t.Parse(templateData)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse template %v", name)
+	}
+	err = t.Execute(&writer, p)
+	return writer.Bytes(), err
+}
+
+// computeCustomizedPrometheusAlerts compute Alerts by merging the data that was get from cephcluster spec with the default data
+func (c *Cluster) computeCustomizedPrometheusAlerts() (map[string]*cephv1.CephAlert, error) {
+	defaultAlerts := make(map[string]*cephv1.CephAlert)
+	fi, err := os.Open(filepath.Clean(path.Join(monitoringPath, defaultPrometheusRuleFile)))
+	if err != nil {
+		return nil, err
+	}
+	err = yaml.NewYAMLToJSONDecoder(fi).Decode(&defaultAlerts)
+	if err != nil {
+		return nil, err
+	}
+	// merge custom alerts values with default values (if any)
+	if c.spec.Monitoring.AlertRuleOverrides != nil {
+		err = mergo.Merge(&defaultAlerts, c.spec.Monitoring.AlertRuleOverrides, mergo.WithOverride)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return defaultAlerts, nil
```

Comment:
> I have a few questions for understanding the requirement:
> 
> 1. What do you mean by "operator post-process"?

After the rule customizations are loaded and applied, I mean that you could apply changes further customizations in code before the prometheus rules are created. For example, here in `computeCustomizedPrometheusAlerts()` seems like the right place to then apply the operator namespace before the CR is returned for creation.

> 2. Is the namespace value that should be filled by the operator is [operatorNamespace](https://github.com/rook/rook/blob/master/pkg/operator/ceph/operator.go#L75)?

Exactly, the operator namespace can be queried from that env var

> 3. Do you want that after the prometheus rule crd was already created, to update the alert in the crd spec to include the operator namespace?

Perhaps this was answered in the first question? The operator namespace could be applied before the rules are created.

Reply:
Alternatively, perhaps a simpler implementation would be a use a variable such as `${operatorNamespace}` inside `prometheusrule-default-values.yaml` that would be a simple string replacement by the operator before merging with the rules.

Reply:
I think we should go with your second suggestion, it might be useful in the future if new alerts that required the operatorNamespace (or any other dynamic value) will be added to the Prometheus Rule, so we can replace all of them at once.

Please let me know which suggestion you prefer.

Reply:
I do like the latter suggestion better for a simple string replace, as you said it's more flexible for future settings as well. Thanks!